packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1870,6 +1870,11 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     if (!decoration.enabled)
       return themeData.disabledColor;
 
+      // Checks if iconTheme.color property is not one of the defaults and returns based on the result
+    if (themeData.iconTheme.color != Colors.black54 && themeData.iconTheme.color != Colors.white70){
```

Comment:
I think you mean Colors.black45, but either way I don't think this is the right approach as other developer who sets themeData.iconTheme.color will sometimes hit the switch case and sometimes not, and if user what to force themeData.iconTheme.color to be Colors.white70, they will never be able to do so due the the switch case.

I will suggest we change InputDecoration class to include the iconTheme.

Reply:
Sorry, it took me a whole month to see this. Can you explain more on this approach? We could just introduce an unselectedColor as an optional parameter that gets set with a default. Wouldn't that be easier?

Reply:
Yes, we should introduce an optional parameter to InputDecoration class that sets the unselected color